add breaking changes in odo 2.2 doc (#4683)
openshift/odo: docs/public/breaking-changes-in-odo-2.2.adoc

. Currently the `wildfly` and `dotnet` component types are not working when converted. We have an issue open for this - https://github.com/openshift/odo/issues/4623 

=== Known bugs and limitations

* https://github.com/openshift/odo/issues/4623
* https://github.com/openshift/odo/issues/4615
* https://github.com/openshift/odo/issues/4594
* https://github.com/openshift/odo/issues/4593

=== Frequently asked questions

* Why odo create URL would fail as it is allowed for devfile ?

It won’t fail in the sense that if you tried the conventional s2i approach and try to create odo url create it would fail with url for 8080 port already present as there would already be one for you. issue - #4621

* Not sure to understand the status of the debug

odo env set DebugPort wont work, instead you would need to use odo config set --env DEBUG_PORT - this is because the s2i to devfile converted devfiles dont have a debug type command defined in them. We would fix this too.

* So every component will now be reported as devfile component but what about existing S2I components

> They should work as is mostly because I tried them and they work, maybe try odo list because that one is quite complex to work across the board.

Reviewer: ```suggestion
They should work as is mostly because we tried them and they work, maybe try `odo list` because it is quite complex to make things work across the board.
```

* If oc based checks in tests are not going to work, is there an alternative odo support around it ?

> I wouldn't say you cannot use oc based checks but they would break because now s2i components being converted to devfile would generate a deployment but the oc would try to find a DeploymentConfig.

Reviewer: ```suggestion
It wouldn't be right to say that you cannot use `oc` based checks, but they would break because now the s2i components are being converted to devfile, odo would generate a Kubernetes `Deployment` but the `oc` would try to find a `DeploymentConfig` on the cluster.
```

Please use `odo push` command to create the component with source deployed
----

would generate a `devfile.yaml` which would be using the S2I images and variables that are part of the `nodejs`.

* *Currently devfile components do not support `--git` and `--binary` components hence we still use S2I component flow when a user wants to create them.*

[source,sh]
----
odo create java --s2i --git ./build.war
Validation
 ✓  Validating component [431ms]

Please use `odo push` command to create the component with source deployed
----
Note - observe there wasn't any conversion step involved here.

* *`odo env set DebugPort` wont work with converted devfile components, instead you would need to use `odo config set --env DEBUG_PORT`.*

. Currently the `wildfly` and `dotnet` component types are not working when converted. We have an issue open for this - https://github.com/openshift/odo/issues/4623 

=== Known bugs and limitations

* https://github.com/openshift/odo/issues/4623
* https://github.com/openshift/odo/issues/4615
* https://github.com/openshift/odo/issues/4594
* https://github.com/openshift/odo/issues/4593

=== Frequently asked questions

> * Why odo create URL would fail as it is allowed for devfile ?

Reviewer: ```suggestion
* Why does odo fail create to URL using `odo url create` for a component created using `odo create --s2i` even though `odo url craete` is allowed for devfile?
```

> It won’t fail in the sense that if you tried the conventional s2i approach and try to create odo url create it would fail with url for 8080 port already present as there would already be one for you. issue - #4621

Reviewer: ```suggestion
It won’t fail in the sense that if you tried the conventional s2i approach and try to create `odo url create` it would fail with url for 8080 port already present as there would already be one for you. issue - #4621
```

> * Is odo env --now and odo delete --wait permanent or temporary as we already have the issue for “normal” devfile components

Reviewer: ```suggestion
* Is `odo env --now` and `odo delete --wait` permanently disabled or temporarily as we already have the issue for “normal” devfile components
```
